components/sup/src/cli.rs

```diff
@@ -44,6 +104,163 @@ mod test {
+                        "hab-sup run --permanent-peer",
+                        "PERMANENT_PEER" => true);
+
+        assert_cli_cmd!(permanent_peer_should_have_short_flag,
+                        "hab-sup run -I",
+                        "PERMANENT_PEER" => true);
+
+        assert_cli_cmd!(should_handle_peer_watch_file,
+                        "hab-sup run --peer-watch-file foobar",
+                        "PEER_WATCH_FILE" => "foobar");
+
+        #[test]
+        fn peer_watch_file_conflicts_with_peer() {
+            let cmd_vec = Vec::from_iter(
+                "hab-sup run --peer-watch-file foobar --peer 1.1.1.1:1111".split_whitespace(),
+            );
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        assert_cli_cmd!(should_handle_ring,
+                        "hab-sup run --ring fizbang",
+                        "RING" => "fizbang");
+
+        assert_cli_cmd!(ring_should_have_short_flag,
+                        "hab-sup run -r flippitygibbit",
+                        "RING" => "flippitygibbit");
+
+        #[test]
+        fn ring_conflicts_with_ring_key() {
+            let cmd_vec =
+                Vec::from_iter("hab-sup run --ring myring --ring-key foobar".split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        assert_cli_cmd!(should_handle_ring_key,
+                        "hab-sup run --ring-key secrit",
+                        "RING_KEY" => "secrit");
+
+        assert_cli_cmd!(should_handle_channel,
+                        "hab-sup run --channel dev",
+                        "CHANNEL" => "dev");
+
+        assert_cli_cmd!(should_handle_bldr_url,
+                        "hab-sup run --url http://not.a.real.com",
+                        "BLDR_URL" => "http://not.a.real.com");
+
+        assert_cli_cmd!(bldr_url_should_have_a_short_flag,
+                        "hab-sup run -u http://not.a.real.com",
+                        "BLDR_URL" => "http://not.a.real.com");
+
+        assert_cli_cmd!(bldr_url_should_have_a_default_value,
+                        "hab-sup run",
+                        "BLDR_URL" => "https://bldr.habitat.sh");
+
+        #[test]
+        fn invalid_bldr_url_should_fail() {
+            let cmd_vec = Vec::from_iter("hab-sup run -u bad-url".split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        // TODO: config dir test
+
+        // TODO: bad config dir test
+
+        assert_cli_cmd!(should_handle_auto_update,
+                        "hab-sup run --auto-update",
+                        "AUTO_UPDATE" => true);
+
+        assert_cli_cmd!(auto_update_should_have_a_short_flag,
+                        "hab-sup run -A",
+                        "AUTO_UPDATE" => true);
+
+        // TODO: key file cert file test
+
+        // TODO: key file requires test
+
+        // TODO: cert file requires test
```

Comment:
The TODO's here all involve options that require interacting with the file system. I wasn't sure the best way to handle that. Suggestions would be appreciated.